stdlib/runtime/stringUtils.gr

```diff
@@ -0,0 +1,172 @@
+// TODO: Remove dependency on Pervasives once Option/Result types are imbedded in the compiler (#1050)
+
+import WasmI32, {
+  add as (+),
+  sub as (-),
+  gtU as (>),
+  geU as (>=),
+  ltU as (<),
+  shrS as (>>),
+  eq as (==),
+  ne as (!=),
+  and as (&),
+} from "runtime/unsafe/wasmi32"
+import WasmI64 from "runtime/unsafe/wasmi64"
+import Memory from "runtime/unsafe/memory"
+import Tags from "runtime/unsafe/tags"
+import { reducedInteger } from "runtime/numbers"
+
+@disableGC
+export let rec parseInt = (string: String, radix: Number) => {
+  let _CHAR_0 = 0x30n
+  let _CHAR_B = 0x42n
+  let _CHAR_b = 0x62n
+  let _CHAR_O = 0x4fn
+  let _CHAR_o = 0x6fn
+  let _CHAR_X = 0x58n
+  let _CHAR_x = 0x78n
+
+  let _CHAR_A = 0x41n
+  let _CHAR_a = 0x61n
+
+  let _CHAR_UNDERSCORE = 0x5fn
+  let _CHAR_MINUS = 0x2dn
+
+  let _INT_MIN = -9223372036854775808N
+
+  // Don't need to process Unicode length since if the string
+  // contains non-ascii characters, it's not a valid integer
+  let strLen = WasmI32.load(WasmI32.fromGrain(string), 4n)
+
+  // Our pointer within the string we're parsing, offset by the
+  // header
+  let mut offset = WasmI32.fromGrain(string) + 8n
+
+  let strEnd = offset + strLen
+
+  let radix = WasmI32.fromGrain(radix)
+  let result = if (WasmI32.eqz(radix & Tags._GRAIN_NUMBER_TAG_MASK) || radix < WasmI32.fromGrain(2) || radix > WasmI32.fromGrain(36)) {
+    Memory.incRef(WasmI32.fromGrain(Err))
+    Err("Radix must be an integer between 2 and 36")
```

Comment:
Do we want exception types for this?

Reply:
Only reason I'm a no is because we don't have a great way to re-export exceptions right now. I think there's an issue for that.

Reply:
Can you open an issue to do this once that issue is solved? Also verify that issue exists 😛

Reply:
https://github.com/grain-lang/grain/issues/1057